test/proxyOFT.ts

```diff
@@ -0,0 +1,721 @@
+import { FakeContract, smock } from "@defi-wonderland/smock";
+import { time } from "@nomicfoundation/hardhat-network-helpers";
+import { expect } from "chai";
+import { ethers, upgrades } from "hardhat";
+import { SignerWithAddress } from "hardhat-deploy-ethers/signers";
+
+import { convertToUnit } from "../helpers/utils";
+import {
+  AccessControlManager,
+  LZEndpointMock,
+  LZEndpointMock__factory,
+  MockToken,
+  MockToken__factory,
+  ResilientOracleInterface,
+  XVS,
+  XVSBridgeAdmin,
+  XVSProxyOFTDest,
+  XVSProxyOFTDest__factory,
+  XVSProxyOFTSrc,
+  XVSProxyOFTSrc__factory,
+  XVS__factory,
+} from "../typechain";
+
+describe("Proxy OFTV2: ", function () {
+  const localChainId = 1;
+  const remoteChainId = 2;
+  const name = "Venus XVS";
+  const symbol = "XVS";
+  const sharedDecimals = 8;
+  const singleTransactionLimit = convertToUnit(10, 18);
+  const maxDailyTransactionLimit = convertToUnit(100, 18);
+
+  let LZEndpointMock: LZEndpointMock__factory,
+    RemoteTokenFactory: XVS__factory,
+    LocalTokenFactory: MockToken__factory,
+    ProxyOFTV2Src: XVSProxyOFTSrc__factory,
+    ProxyOFTV2Dest: XVSProxyOFTDest__factory,
+    localEndpoint: LZEndpointMock,
+    remoteEndpoint: LZEndpointMock,
+    localOFT: XVSProxyOFTSrc,
+    remoteOFT: XVSProxyOFTDest,
+    localToken: MockToken,
+    bridgeAdminRemote: XVSBridgeAdmin,
+    bridgeAdminLocal: XVSBridgeAdmin,
+    remoteToken: XVS,
+    localPath: string,
+    acc2: SignerWithAddress,
+    acc3: SignerWithAddress,
+    acc1: SignerWithAddress,
+    accessControlManager: FakeContract<AccessControlManager>,
+    oracle: FakeContract<ResilientOracleInterface>,
+    defaultAdapterParams: any;
+
+  before(async function () {
+    LZEndpointMock = await ethers.getContractFactory("LZEndpointMock");
+    ProxyOFTV2Src = await ethers.getContractFactory("XVSProxyOFTSrc");
+    ProxyOFTV2Dest = await ethers.getContractFactory("XVSProxyOFTDest");
+    LocalTokenFactory = await ethers.getContractFactory("MockToken");
+    accessControlManager = await smock.fake<AccessControlManager>("AccessControlManager");
+    accessControlManager.isAllowedToCall.returns(true);
+    RemoteTokenFactory = await ethers.getContractFactory("XVS");
+    acc1 = (await ethers.getSigners())[0];
+    acc2 = (await ethers.getSigners())[1];
+    acc3 = (await ethers.getSigners())[2];
+    oracle = await smock.fake<ResilientOracleInterface>("ResilientOracleInterface");
+    oracle.getPrice.returns(convertToUnit(1, 18));
+    defaultAdapterParams = ethers.utils.solidityPack(["uint16", "uint256"], [1, 200000]);
+  });
+
+  beforeEach(async function () {
+    localEndpoint = await LZEndpointMock.deploy(localChainId);
```

Comment:
These deployments could be saved as a fixture in the before hook and then returned in each test instead of using this hook

Reply:
[aba9539](https://github.com/VenusProtocol/token-bridge/pull/1/commits/aba9539f61aa34f2189e28601e2e2bbcd09166b0)